plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/filtering/SemanticFilterRuleUtil.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package de.cau.cs.kieler.klighd.filtering;
+
+/**
+ * Util for easily getting some rules.
+ * 
+ * @author tik
+ *
+ */
+public abstract class SemanticFilterRuleUtil {
+    /** An empty tag used to define constants. */
+    private static final SemanticFilterTag EMPTY = new SemanticFilterTag("");
+    /** A rule that is always evaluated to false. */
+    public static final SemanticFilterRule FALSE =
+            new AndConnective(EMPTY, new NegationConnective(EMPTY));
+    /** A rule that is always evaluated to true. */
+    public static final SemanticFilterRule TRUE =
+            new OrConnective(EMPTY, new NegationConnective(EMPTY));
+
+    /**
+     * Constructs an unnamed rule with all operands connected via OR. If no operands are given, this
+     * is always false.
+     * 
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigOrConnective(SemanticFilterRule... operands) {
+        return getBigOrConnective(null, operands);
+    }
+
+    /**
+     * Constructs a named rule with all operands connected via OR. If no operands are given, this is
+     * always false.
+     * 
+     * @param ruleName
+     *            The rule name.
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigOrConnective(String ruleName,
+            SemanticFilterRule... operands) {
+        if (operands.length <= 0) {
+            // No rules, automatically evaluate to false
+            return FALSE;
+        } else if (operands.length <= 1) {
+            // Just one rule, true iff rule is true
+            // Add ruleName instead of just using first operand twice to avoid double evaluation
+            return addRuleName(ruleName, operands[0]);
+        }
+        OrConnective bigOr = new OrConnective(operands[0], operands[1], ruleName);
+        for (int i = 2; i < operands.length; i++) {
+            bigOr.rightOperand = new OrConnective(bigOr.rightOperand, operands[i]);
+        }
+        return bigOr;
+    }
+
+    /**
+     * Constructs an unnamed rule with all operands connected via AND. If no operands are given,
+     * this is always false.
+     * 
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigAndConnective(SemanticFilterRule... operands) {
```

Comment:
this is fine to leave here as a utility, but it is probably not going to be used, because the checkboxes in the UI will serve as the ANDs, at least for application of filters in the proxy view context. In the future someone might of course think of something completely different to use the filters for

Reply:
It also could just be a convenience kinda thing. Maybe someone can think of a sensible rule requiring multiple ands